diskmaker/controllers/lv/reconcile.go

```diff
@@ -415,6 +426,16 @@ func (r *LocalVolumeReconciler) Reconcile(ctx context.Context, request ctrl.Requ
 					common.LocalVolumeOwnerNamespaceForPV: r.localVolume.Namespace,
 				}
 
+				if createSymlinkStatus == SymlinkCreated && deviceNameLocation.forceWipe {
```

Comment:
What if a diskmaker process is restarted after `r.createSymlink` above and before this line? The previous diskmaker created the symlink, but did not have time to wipe the volume, and the new one found the existing symlink and did not wipe the volume either, because it got `SymlinkExists`.

IMO, diskmaker should wipe volumes *before* creating the symlink to have it bulletproof. In the worst case, it will be wiped several times, but it will be wiped before PV is created.

Reply:
Good catch, fully agreed!